packages/nx/src/project-graph/nx-deps-cache.ts

```diff
@@ -99,7 +100,13 @@ export function createCache(
   return newValue;
 }
 
-export function writeCache(cache: ProjectGraphCache): void {
+export function writeCacheIfNecessary(
+  cache: ProjectGraphCache,
+  existingCache: ProjectGraphCache
+): void {
+  if (deepEquals(cache, existingCache)) {
```

Comment:
Not sure why this was exposed with this PR exactly but the workers were trying to read and write the project graph cache at the same time... so I just added a check to not write unless necessary and that fixed the problem. Seems kinda like a bandaid though